src/psyclone/gocean1p0.py

```diff
@@ -1286,6 +1379,86 @@ def gen_arg_setter_code(self, parent):
                 sub, "check_status",
                 ["'clSetKernelArg: arg {0} of {1}'".format(index, self.name),
                  err_name]))
+            index = index + 1
+
+    def _prepare_opencl_kernel_schedule(self):
+        ''' GOcean OpenCL kernels take the iteration boundaries as arguments
+        ans adds a conditional masking statement to avoid updating elements
+        outside the boundaries.
+        '''
+        kschedule = self.get_kernel_schedule()
+
+        # If the first statement in the kernel schedule is not marked
```

Comment:
Ops no, I dropped this functionality because even kernels that appear several times are just generated once in OpenCL, so the conditional mask check statement is always inserted just once as expected. I deleted the comment now.